### PR TITLE
feat(DatePicker): Added DatePicker component

### PR DESCRIPTION
**Pull request by HHogg**
datepicker

### CODE REVIEW

**Comment by thatguynamedandy** [comment]
Nice work @HHogg this is a meaty task made to look pretty easy.

File: src/components/date-picker/DatePickerContext.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+  handleActiveStartDateChange(direction) {
+    const activeStartDate = changeMonth(this.state.activeStartDate, direction);
+    let activeEndDate = this.state.activeEndDate;
+
+    if (isSameMonth(activeStartDate, activeEndDate)) {
+      activeEndDate = changeMonth(activeEndDate, 1);
+    }
+
+    this.setState({
```
[the remark this answers is not in the document]
Comment (HHogg) [reply src/components/date-picker/DatePickerContext.js]: Good shout.

File: src/components/date-picker/DatePickerContext.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+import DatePickerControls from './DatePickerControls';
+import DatePickerHeaderControl from './DatePickerHeaderControl';
+import DatePickerViewMonth from './DatePickerViewMonth';
+import {
+  dateOrNow,
+  changeMonth,
+  isBeforeDay,
+  isAfterDay,
+  isSameMonth,
+} from './_utils';
```
[the remark this answers is not in the document]
Comment (HHogg) [reply src/components/date-picker/DatePickerContext.js]: I'm not sure ... there's another util function for Position that is prefixed with _. I've updated as it's not really needed. 👍

**Comment by pmsorhaindo** [comment]
Woah!!

**Pull request merged**